### ISSUE TITLE
Remember group expansion with dynamic data, or hide groups?

### ISSUE THREAD

**Issue by prefect4742** [h6ah4i/android-advancedrecyclerview#346]
I have a list of groups that can come and go. I handle this with flags inside my data model. When the data model indicates that groups have become visible I call notifyDataSetChanged on my AbstractExpandableItemAdapter subclass, since I don't (yet) want to deal with the complexity if inserting/removing groups and so everything is recreated.

My model groups also save if the group was expanded or not. How do I expand the group when onBindGroupViewHolder() is called? Or is there another callback that I can do it in?

If I try to call expandGroup() in onBindGroupViewHolder() I get an IllegalStateException ("Cannot call this method while RecyclerView is computing a layout or scrolling").

Alternatively, is there a way to set the visibility of a group so that it is not included in the list but still allocated?

**Comment by h6ah4i** [comment]
Hi. Thanks for a good question.

I think there is no clean way to achieve such behavior. Maybe, `Handler` can be used for workaround, but that is not straightforward.

> My model groups also save if the group was expanded or not. How do I expand the group when onBindGroupViewHolder() is called? Or is there another callback that I can do it in?

Okay, I've created a new branch to test a new feature solve the problem. I added `getInitialGroupExpandedState()` method to control initial state of group items.

```java
class MyExpandableAdapter extends AbstractExpandableItemAdapter<...> {
    @Override
    public boolean getInitialGroupExpandedState(int groupPosition) {
        return /* true or false */;
    }
    ...
}
```

To use the test branch code via JitPack: 

```gradle
repositories {
    maven { url "https://jitpack.io" }
}
dependencies {
    compile 'com.github.h6ah4i:android-advancedrecyclerview:e1b38cb892206461ca11dfb2cb017beca92ccd82'
}
```

https://github.com/h6ah4i/android-advancedrecyclerview/commit/e1b38cb892206461ca11dfb2cb017beca92ccd82

**Comment by prefect4742** [comment]
It seems to work, however if I expand one group and (maybe, not entirely sure) later call notifyDataSetChanged(), I get calls to getGroupItemViewType() with a position of -1. Is that expected? I haven't seen it happen without this change.

**Comment by h6ah4i** [comment]
Oh, that seems a bug, so I will take a look at it. Thanks.

**Comment by h6ah4i** [comment]
@prefect4742 Hi, I have fixed the `getInitialGroupExpandedState ()` bug. Could you try the latest code via JitPack?

```gradle
repositories {
    maven { url "https://jitpack.io" }
}
dependencies {
    compile 'com.github.h6ah4i:android-advancedrecyclerview:36a72c48a58e804505d6d22f83025f47f499332d'
}
```

**Comment by h6ah4i** [comment]
@prefect4742 This feature has been merged and released as `v0.10.4` now.

**Comment by prefect4742** [comment]
Hi, sorry for taking a while. I have tested and it seems to work as expected now :)
